### DISCUSSION TITLE
Firebase로그인 과정에서의 데이터 전달에 관한 질문입니다.

### DISCUSSION THREAD

**Discussion by park7628** [park7628/ARUnityProject#51] [Q&A]
### 데이터를 전달받는 과정에서 의문이 생겼는데 해결 방안을 몰라 글을 남깁니다.

![image](https://github.com/park7628/ARUnityProject/assets/108246145/999e3535-e2a2-4be9-89d7-0ba9f0254932)
다음 코드는 UI에서 활성화 비활성화를 통해 화면을 전환하는 코드입니다.

![image](https://github.com/park7628/ARUnityProject/assets/108246145/8ca68fea-abe2-4505-accf-95b3ed4a89aa)
다음 코드는 FIrebase에서 로그인 처리를 하는 코드입니다.

![image](https://github.com/park7628/ARUnityProject/assets/108246145/c8de1d4c-03ed-413c-8bc3-bebe45cbfc08)
실행을 시키고 알맞은 이메일과 PW를 입력하면 다음과 같이 나옵니다.
버튼을 클릭하면 Experiment1SetActive()가 실행 되도록 했습니다.

순서 상으로는 if문을 돌기 전 Firbase로 넘어가서 먼저 로그인 처리를 해야 되는데 Experiment1SetActive()에 있는 if문이 먼저 실행됩니다. 이를 해결하기 위해 FIrbase에서  Experiment1SetActive()로 넘어가게 해봤지만 안됐습니다. 
![image](https://github.com/park7628/ARUnityProject/assets/108246145/c8376c44-5944-4d0b-b6ca-a10d2ccc5195)

bool 값을 static으로 선언해서 Firebase에서 바꿔준 뒤 Experiment1SetActive()에서 if문을 실행시켜 봤습니다. static으로 선언했던 bool값들은 잘 바뀌어 if문이 잘 실행되는 것 까지는 확인했으나,  SetActive()함수는 실행되지 않는 현상을 확인했습니다.

그래서 아예 Firebase스크립트에서 SetActive해줘야 되는 UI를 받아 직접 바꿔주는 방법을 했었지만 바뀌지 않았습니다.
![image](https://github.com/park7628/ARUnityProject/assets/108246145/73caafeb-1c2e-447f-b9e3-031b8e7af81c)

그리고 비동기 방식을 사용해 async와 await를 사용해서 실행해봤는데 항상 마지막에 FirebaseLogInManager의 Login()이 실행 되는 현상을 확인하고 해결 방법을 모르겠어 남깁니다.

@jongfeel

**Comment by jongfeel** [comment]
안녕하세요.

이건 비동기 프로그램 동작 방식을 이해하지 못해서 생기는 문제라고 보여집니다.
제가 한번 살펴보고 필요한 질문을 남겨 보겠습니다.

**Reply by park7628** [reply]
감사합니다!!
